Foundation/NSHTTPCookie.swift

```diff
@@ -459,3 +554,19 @@ public class HTTPCookie : NSObject {
         return _portList
     }
 }
+
+//utils for cookie parsing
+internal extension String {
```

Comment:
A style note, but I'd rather just put these functions inline at the call site than extending String in the entire project.